src/pages/search/models.tsx

```diff
@@ -0,0 +1,156 @@
+      <Container fluid>
+        <Stack
+          sx={(theme) => ({
+            height: 'calc(100vh - 2 * var(--mantine-header-height,50px))',
+            position: 'fixed',
+            left: 0,
+            top: 'var(--mantine-header-height,50px)',
+            width: '377px',
+            overflowY: 'auto',
+            padding: theme.spacing.md,
+          })}
+        >
+          <RenderFilters />
+        </Stack>
+
+        <Stack pl={377} w="100%">
+          <SearchHeader />
+          <ModelsHitList />
+        </Stack>
+      </Container>
+    </InstantSearch>
+  );
+}
+
+const RenderFilters = () => {
+  return (
+    <>
+      <SortBy
+        title="Sort models by"
+        items={[
+          { label: 'Highest Rated', value: 'models:metrics.weightedRating:desc' },
+          { label: 'Most downloaded', value: 'models:metrics.downloadCount:desc' },
+          { label: 'Most Liked', value: 'models:metrics.favoriteCount:desc' },
+          { label: 'Most discussed', value: 'models:metrics.commentCount:desc' },
+          { label: 'Newest', value: 'models:createdAt:desc' },
+        ]}
+      />
+      <ChipRefinementList
+        title="Filter by Base Model"
+        attribute="modelVersion.baseModel"
+        sortBy={['name']}
+      />
+      <ChipRefinementList title="Filter by Model Type" attribute="type" sortBy={['name']} />
+      <ChipRefinementList
+        title="Filter by Checkpoint Type"
+        sortBy={['name']}
+        attribute="checkpointType"
+      />
+      <SearchableMultiSelectRefinementList
+        title="Tags"
+        attribute="tags"
+        // TODO.Search: Meiliserach & facet searching is not supported when used with sort by as Angolia provides it.  https://github.com/meilisearch/meilisearch-js-plugins/issues/1222
+        // If that ever gets fixed, just make sortable true + limit 20 or something
+        limit={9999}
+        searchable={false}
+      />
+    </>
+  );
+};
+
+const useStyles = createStyles((theme) => ({
+  grid: {
+    display: 'grid',
+    gridTemplateColumns: `repeat(auto-fill, minmax(250px, 1fr))`,
+    columnGap: theme.spacing.md,
+    gridTemplateRows: `auto 1fr`,
+    overflow: 'hidden',
+    marginTop: -theme.spacing.md,
+
+    '& > *': {
+      marginTop: theme.spacing.md,
+    },
+  },
+}));
+
+export function ModelsHitList() {
+  const { hits, showMore, isLastPage } = useInfiniteHits();
+  const { status } = useInstantSearch();
+  const { ref, inView } = useInView();
+  const { classes } = useStyles();
+
+  // #region [infinite data fetching]
+  useEffect(() => {
+    if (inView && status === 'idle' && !isLastPage) {
+      showMore?.();
+    }
+  }, [status, inView, showMore, isLastPage]);
+
+  return (
+    <Stack>
+      <Box className={classes.grid}>
+        {hits.map((hit) => {
+          const modelHit = hit as unknown as ModelGetAll['items'][number];
+          const images = (hit.images ?? []) as ModelGetAll['items'][number]['image'][];
```

Comment:
We can avoid these castings if we explicitly set the generic in `useInfiniteHits`